### PR TITLE
Added doctest for TensorBlock usage and add_gradient.

### PR DESCRIPTION
**Pull request by fgrassel**
----
:books: Documentation preview :books:: https://equistore--204.org.readthedocs.build/en/204/

### CODE REVIEW

**Review by DavideTisi — APPROVED** [review]
Very nice works, I think it will be very useful for anyone that wants to start creating a TensorBlock.

File: python/src/equistore/block.py [hunk 1]
```diff
@@ -26,6 +26,70 @@ class TensorBlock:
     A block can also contain gradients of the values with respect to a variety
     of parameters. In this case, each gradient has a separate set of samples,
     and possibly components but share the same property labels as the values.
+
+    This is an example on how to:
+      - create a ``TensorBlock``
+      - add a gradient via ``add_gradient``
+      - access values with a specific label value, by using the ``position``
+        function
```
Comment (Luthaf) [inline python/src/equistore/block.py:34]: This is a nice example, but I'm not sure about the organization of the code. I tend to prefer short and focused examples inside docstrings, and move examples combining multiple functions to short tutorials/how-to.

Concretely, I would only leave the part constructing a TensorBlock here, and make the rest of this example into two how-to: "manipulating gradients on blocks" and "accessing specific data in blocks"

**Comment by Luthaf** [comment]
Rebased and adapted to the latest version, let's merge this!

**Review by DavideTisi — APPROVED** [review]
great perfect for me

**Pull request merged**